core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -116,31 +102,33 @@ private void reportCreationError(Exception ex, Uri videoUri) {
     private final MediaPlayer.OnVideoSizeChangedListener internalSizeChangedListener = new MediaPlayer.OnVideoSizeChangedListener() {
         @Override
         public void onVideoSizeChanged(MediaPlayer mp, int width, int height) {
-            if (onSizeChangedListener == null) {
-                return;
+            if (onVideoSizeChangedForwarder == null) {
+                throw new IllegalStateException("Should bind a OnVideoSizeChangedListener. Cannot forward events.");
```

Comment:
this is from https://github.com/novoda/no-player/pull/38 for now if we only support videos it makes sense to me